From: Stephen Boyd <sboyd@codeaurora.org>
Date: Wed, 26 Oct 2016 00:35:50 -0700
Subject: clocksource/drivers/arm_arch_timer: Map frame with
 of_io_request_and_map()
Git-commit: f947ee147e0895d5c0ef18e81ca7a4e4219739e3
Patch-mainline: v4.10-rc1
References: fate#322003

Let's use the of_io_request_and_map() API so that the frame
region is protected and shows up in /proc/iomem.

Signed-off-by: Stephen Boyd <sboyd@codeaurora.org>
Signed-off-by: Daniel Lezcano <daniel.lezcano@linaro.org>
Acked-by: Marc Zyngier <marc.zyngier@arm.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/clocksource/arm_arch_timer.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/clocksource/arm_arch_timer.c
+++ b/drivers/clocksource/arm_arch_timer.c
@@ -914,8 +914,9 @@ static void __init arch_timer_mem_init(s
 		best_frame = of_node_get(frame);
 	}
 
-	base = arch_counter_base = of_iomap(best_frame, 0);
-	if (!base) {
+	base = arch_counter_base = of_io_request_and_map(best_frame, 0,
+							 "arch_mem_timer");
+	if (IS_ERR(base)) {
 		pr_err("arch_timer: Can't map frame's registers\n");
 		of_node_put(best_frame);
 		return;
